image.h

```diff
@@ -0,0 +1,20 @@
+#include <cstdint>
+#include <memory>
+
+class Image {
+public:
+    Image(uint32_t rows, uint32_t columns);
+    Image(const Image& image); // Copy constructor
+    ~Image();
+
+    uint32_t GetRows();
+    uint32_t GetColumns();
+    
+    uint32_t GetPixelValue(uint32_t pixelIndex);

```

Comment:
Good idea to do pixel manipulation via these functions. Whenever you make get and set functions, look for an opportunity for encapsulation - opportunity to keep functionality together, so that changes can be localized and not spread out.
Here the questions I can think of now:
1. Is it better to have `pixelIndex` or `x` and `y` as the parameters to these get and set functions?
2. For what will clients get and set a single pixel? Or will they only do variety of operations on the whole image?
3. If I change the pixel representation (e.g., keep in a file and load into memory only when pixel operations are required), will the clients need to change?

Reply:
The pixelIndex approach will be useful for the current scenario of "applying changes to whole image" as there wont be calculations required inside each getter and setter . 
However this will not be good approach when we need modifications on  selected set of pixel . 
So changed to (x,y) as suggested in review .